File: movie/lib/model/movie_model.dart
import 'package:movie/model/favorite_model.dart';

class movie_model {
  final int id;
  final String title;
  final String description;
  final String image;
  final int subcategories_id;
  // List<favorite_model> favorite;
  movie_model({
    required this.id,
    required this.title,
    required this.description,
    required this.image,
    required this.subcategories_id,
    // required this.favorite,
  });
  factory movie_model.fromjson(Map<String, dynamic> json) {
    return movie_model(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      image: json['image'],
      subcategories_id: json['subcategories_id'],
      // favorite: (json['favorite'] as List).map((favorite) => favorite_model.fromjson(favorite)).toList(),
    );
  }
}

File: movie_back/app/Models/movie_model.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class movie_model extends Model
{
    use HasFactory;
    protected $table = "movie";
    protected $primaryKey = "id";
    protected $fillable = [
        'title',
        'description',
        'status',
        'image',
        'video',
        'subcategories_id',
    ];
    public function subcategories(){
        return $this->belongsTo(subcategories_model::class);
    }
    public function favorite(){
        return $this->hasMany(favorite_model::class,'movie_id');
    }
}
